src/main/java/no/nav/veilarbperson/service/AuthService.java

```diff
@@ -75,19 +99,68 @@ public void sjekkLesetilgang(Fnr fnr) {
     public boolean harLesetilgang(Fnr fnr) {
         AktorId aktorId = aktorOppslagClient.hentAktorId(fnr);
 
-        if (erEksternBruker()) {
-            return veilarbPep.harTilgangTilPerson(getInnloggetBrukerToken(), ActionId.READ, aktorId);
-        } else if (erInternBruker()) {
-            return veilarbPep.harVeilederTilgangTilPerson(authContextHolder.requireNavIdent(), ActionId.READ, aktorId);
-        } else {
-            if (!erSystemBruker()) {
-                throw new ResponseStatusException(HttpStatus.FORBIDDEN);
-            }
-            return harAADRolleForSystemTilSystemTilgang()
-                    || veilarbPep.harTilgangTilPerson(getInnloggetBrukerToken(), ActionId.READ, aktorId);
-        }
+		if (unleashClient.isEnabled(UNLEASH_POAO_TILGANG_ENABLED)) {
+			if (erEksternBruker()) {
+				return harEksternBrukerTilgangTilEksternBruker(fnr.get());
+			} else if (erInternBruker()) {
+				return harVeilederTilgabgTilEksternBruker(fnr.get());
+			} else {
+				if (!erSystemBruker()) {
+					throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+				}
+				return harAADRolleForSystemTilSystemTilgang()
+						|| harEksternBrukerTilgangTilEksternBruker(fnr.get());
+			}
+		} else {
+			if (erEksternBruker()) {
+				return veilarbPep.harTilgangTilPerson(getInnloggetBrukerToken(), ActionId.READ, aktorId);
+			} else if (erInternBruker()) {
+				return veilarbPep.harVeilederTilgangTilPerson(authContextHolder.requireNavIdent(), ActionId.READ, aktorId);
+			} else {
+				if (!erSystemBruker()) {
+					throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+				}
+				return harAADRolleForSystemTilSystemTilgang()
+						|| veilarbPep.harTilgangTilPerson(getInnloggetBrukerToken(), ActionId.READ, aktorId);
+			}
+		}
     }
 
+	private boolean harEksternBrukerTilgangTilEksternBruker(String ressursNorskIdent) {
+		Decision desicion = poaoTilgangClient.evaluatePolicy(new EksternBrukerTilgangTilEksternBrukerPolicyInput(
+				authContextHolder.getUid().orElseThrow(), ressursNorskIdent
+		)).get();
```

Comment:
Det kommer ikke så godt frem i Java-kode, men .get() returnerer null hvis requestet til poao-tilgang feiler. Det er bedre å heller bruke getOrThrow() som vil kaste exceptionet fra poao-tilgang videre istedenfor for at det blir kastet NPE. Samme gjelder også linje 143.

Reply:
Done :)